### ISSUE TITLE
`(CaseClauseError) no case clause matching: nil` for some lookups

### ISSUE THREAD

**Issue by OldhamMade** [mathieuprog/tz#8]
```
iex(1)> DateTime.now!("America/Argentina/Salta", Tz.TimeZoneDatabase)
** (CaseClauseError) no case clause matching: nil
    (tz 0.20.0) lib/time_zone_database.ex:43: Tz.TimeZoneDatabase.do_find_period_for_secs/3
    (tz 0.20.0) lib/time_zone_database.ex:33: Tz.TimeZoneDatabase.find_period_for_secs/3
    (elixir 1.12.0) lib/calendar/datetime.ex:623: DateTime.shift_zone_for_iso_days_utc/5
    (elixir 1.12.0) lib/calendar/datetime.ex:736: DateTime.now!/2
```

This is with Elixir 1.12.0 and Erlang 24.0 on macOS.

**Comment by mathieuprog** [comment]
Hi! Will look at it thank you!

**Comment by mathieuprog** [comment]
> iex(1)> DateTime.now!("America/Argentina/Salta", Tz.TimeZoneDatabase)
> #DateTime<2021-09-21 09:45:52.535000-03:00 -03 America/Argentina/Salta>

> elixir --version
> Erlang/OTP 22 [erts-10.7] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1]
> 
> Elixir 1.11.3 (compiled with Erlang/OTP 21)

Any thoughts? Can you run it now?

**Comment by mathieuprog** [comment]
The error is cause by the option `:reject_time_zone_periods_before_year`. Removing this option will prevent the error.
That will be a quick solution before I fix it.
Note that I don't even know if this option is worth it with the newer version of Tz because I think it might make no difference memory-wise. It needs to be measured (I just don't know how yet). Also this option was more intended for embedded devices that have a very limited amount of memory.

**Comment by OldhamMade** [comment]
Ah yes, apologies, I should have mentioned that I'm using that option. 

It's the main reason I'm looking to switch to Tz: right now I'm using Tzdata but I'm running in a memory-restricted environment. I'm looking for something with a smaller memory footprint, and I only care about dates in the future, so Tz with the reject option looked perfect for my needs.

**Comment by mathieuprog** [comment]
Are you able to measure the memory footprint?

**Comment by mathieuprog** [comment]
Fixed in 0.20.1

**Comment by mathieuprog** [comment]
Still interested to know if you can measure memory footprint:)
